app/models/concerns/reports/web_hook_events_daily_aggregate.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module Reports::WebHookEventsDailyAggregate
+  extend ActiveSupport::Concern
+  class_methods do
+    def report_web_hook_events_daily_aggregate(report)
+      report.labels = [
+        { type: :date, property: :x, title: I18n.t("reports.default.labels.day") },
+        { type: :percent, property: :y, title: I18n.t("reports.default.labels.percent") },
+      ]
+
+      type = report.filters.dig(:type_of_web_hook_event)
+      # all = failed_events + successful_events per day;
+      # failed events = failed events per day;
+      # successful events = successful events per day;
+      # mean duration = mean duration of all events per day;
+      report.add_filter(
+        "type_of_web_hook_event",
+        type: "list",
+        default: type || "all",
```

Comment:
Looks like to get rid of "any" in the dropdown you could do `allow_any: false` here?

https://github.com/discourse/discourse/blob/38b83941488a25dca11d042b1e4e1b16a52c832e/app/assets/javascripts/admin/addon/components/report-filters/list.hbs#L6-L6

Reply:
Found this component based on this:

https://github.com/discourse/discourse/blob/38b83941488a25dca11d042b1e4e1b16a52c832e/app/assets/javascripts/admin/addon/components/admin-report.hbs#L193-L212

Reply:
<img width="241" alt="image" src="https://github.com/discourse/discourse/assets/70247653/db7663a7-7e52-4e95-94ca-5fe082d9b405">


Noice! I had to add an `auto_insert_none_item` option, but I got it to work! Thanks

> I am not sure what you mean by this, can you please add some more detail? What text filter do you need?

I wanted a generic plain text field so that the user could add the `payload_url` from the web_hook to it, and we could filter it more precisely. But the more I think about it, the more I wonder if it will be necessary for my case; the user can use the same `payload_url` for everything, for example, and they do not have easy access to the web_hook_id (they can get it by the URL but I'm not sure if that would be a great UX for the report page).

I would need to add to that `reports-filters` dir a `text/string.hbs` file, but given this, I'm unsure if it makes sense.

Reply:
Ah right I see what you mean now, since the aggregates are stored against each webhook you want to add further filtering using the webhook ID/URL. I am not totally sure about this...maybe just ship this as is first and go from there?